// react-giphy-searchbox/lib/components/SearchForm/SearchForm.js
'use strict';

exports.__esModule = true;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _SearchFormModule = require('./SearchForm.module.css');

var _SearchFormModule2 = _interopRequireDefault(_SearchFormModule);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SearchForm = function SearchForm(_ref) {
  var onSubmit = _ref.onSubmit,
      placeholder = _ref.placeholder,
      searchFormClassName = _ref.searchFormClassName,
      setValue = _ref.setValue,
      value = _ref.value;
  return _react2.default.createElement(
    'form',
    {
      'data-testid': 'SearchFormForm',
      onSubmit: onSubmit,
      autoComplete: 'off',
      className: '' + _SearchFormModule2.default.form + (searchFormClassName ? ' ' + searchFormClassName : '')
    },
    _react2.default.createElement('input', {
      'data-testid': 'SearchFormInput',
      type: 'text',
      placeholder: placeholder,
      onChange: setValue,
      value: value,
      name: 'search',
      className: _SearchFormModule2.default.input
    })
  );
};

exports.default = SearchForm;
module.exports = exports['default'];
