// react-giphy-searchbox/lib/components/ImageItem/ImageItem.js
'use strict';

exports.__esModule = true;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ImageItemModule = require('./ImageItem.module.css');

var _ImageItemModule2 = _interopRequireDefault(_ImageItemModule);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ImageItem = function ImageItem(_ref) {
  var backgroundColor = _ref.backgroundColor,
      item = _ref.item,
      size = _ref.size,
      listItemClassName = _ref.listItemClassName,
      onSelect = _ref.onSelect;
  return _react2.default.createElement(
    'button',
    {
      'data-testid': 'ImageItemButton',
      type: 'button',
      className: '' + _ImageItemModule2.default.imageButton + (listItemClassName ? ' ' + listItemClassName : ''),
      style: {
        backgroundColor: backgroundColor,
        width: size + 'px',
        height: item.images.fixed_width_downsampled.height * size / item.images.fixed_width_downsampled.width + 'px'
      },
      onClick: function onClick() {
        return onSelect(item);
      }
    },
    _react2.default.createElement('img', {
      'data-testid': 'ImageItemImage',
      width: item.images.fixed_width_downsampled.width,
      height: item.images.fixed_width_downsampled.height,
      alt: item.title,
      src: item.images.fixed_width_downsampled.url,
      className: _ImageItemModule2.default.image
    })
  );
};

exports.default = ImageItem;
module.exports = exports['default'];
